### PR TITLE
To mix or not to mix [wip]

### PR DESCRIPTION
**Pull request by sebastiandedeyne**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,11 +12,10 @@ _ide_helper*.php
 /node_modules
 /npm-debug.log
 /yarn-error.log
-/public/build
-/resources/assets/css
+/public/doc/sass
 /public/css
```
Comment (sebastiandedeyne) [inline .gitignore:16]: Resources aren't in a `build` folder anymore, but immediately in `public`, so I split them in `js` and `css`.

File: .gitignore [hunk 2]
```diff
@@ -12,11 +12,10 @@ _ide_helper*.php
...
 /public/js
-/public/doc/sass
+/public/mix-manifest.json
```
Comment (sebastiandedeyne) [inline .gitignore:18]: `build/manifest.json` is now `mix-manifest.json`

File: package.json [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
+    "babel-preset-es2015": "^6.22.0",
+    "babel-preset-es2015-native-modules": "^6.9.4",
     "babel-preset-latest": "^6.16.0",
-    "blender-gulp": "^3.0.0",
     "blender-media": "^3.3.2",
     "blender.js": "^1.3.0",
     "browsernizr": "1.0.2",
     "chart.js": "^1.0.2",
-    "css-loader": "^0.26.0",
-    "dom-storage": "^2.0.2",
```
Comment (sebastiandedeyne) [inline package.json:27]: Cleaned up some old deps

File: package.json [hunk 4]
```diff
@@ -1,42 +1,46 @@
...
-    "dotenv": "^1.2.0",
     "eslint": "^3.7.1",
     "eslint-config-spatie": "^1.2.7",
     "gulp": "^3.9.0",
+    "import-glob-loader": "^1.1.0",
```
Comment (sebastiandedeyne) [inline package.json]: The only thing we use that Laravel Mix doesn't provide out of the box.

File: package.json [hunk 5]
```diff
@@ -1,42 +1,46 @@
...
     "jquery": "^3.0.0",
+    "laravel-mix": "^0.5.16",
     "lodash": "^4.16.6",
+    "node-noop": "^1.0.0",
```
Comment (sebastiandedeyne) [inline package.json:36]: Needed to ignore image imports. Minor hack ¯\\\_(ツ)\_/¯

File: resources/assets/js/back/app.js [hunk 6]
```diff
@@ -15,27 +15,23 @@ import { query } from 'spatie-dom';
 if (query('blender-media')) {
     require.ensure([], () => {
         require('./modules/media').default();
-    }, 'back.media');
+    });
```
Comment (sebastiandedeyne) [inline resources/assets/js/back/app.js]: These were pretty useless anyway.

File: resources/assets/sass/back/back.scss [hunk 7]
```diff
@@ -1,4 +1,4 @@
 @charset 'UTF-8';
 
-@import '@spatie/blender-css/dist/blender';
+@import 'node_modules/@spatie/blender-css/dist/blender';
```
Comment (sebastiandedeyne) [inline resources/assets/sass/back/back.scss:3]: This way it works out of the box. Don't think it's worth the extra config for the edge case of importing from `node_modules`. Yay / nay?

File: resources/assets/sass/front/helpers/above-below.scss [hunk 8]
```diff
@@ -1,38 +1,38 @@
 /* Above */
 
-&.h-above-s {
+.h-above-s {
```
Comment (sebastiandedeyne) [inline resources/assets/sass/front/helpers/above-below.scss]: 🐛

**Comment by sebastiandedeyne** [comment]
Laravel Mix!

## Pros

- Standard Laravel way
- Less stuff for us to maintain (only kinda though, if something breaks or we have an edge case, we still need the know-how)

## Cons

- It's not perfect for our use case, lots of moving parts (way more than we need)—and our own implementation would be pretty light weight (if we'd handle revisioning server-side)
- Possible future issue: postcss currently isn't supported for extension

Thoughts? Yay or nay?

**Review by freekmurze — COMMENTED** [review]
Yay for me. There's a lot of value in just following Laravel here.

File: package.json [hunk 9]
```diff
@@ -1,42 +1,44 @@
...
     "blender-media": "^3.3.2",
     "blender.js": "^1.3.0",
     "browsernizr": "1.0.2",
     "chart.js": "^1.0.2",
-    "css-loader": "^0.26.0",
-    "dom-storage": "^2.0.2",
-    "dotenv": "^1.2.0",
     "eslint": "^3.7.1",
     "eslint-config-spatie": "^1.2.7",
     "gulp": "^3.9.0",
```
Comment (willemvb) [inline package.json]: Gulp be gone?

File: resources/assets/sass/back/back.scss (hunk 7 above)
Comment (willemvb) [reply resources/assets/sass/back/back.scss:3]: Yes

File: resources/assets/js/back/app.js [hunk 10]
```diff
@@ -15,27 +15,23 @@ import { query } from 'spatie-dom';
...
     require.ensure([], () => {
         require('./modules/chart').default();
-    }, 'back.chart');
+    });
 }
 
-/*
-*  uncomment when redactor is installed
-*
 if (query('[data-editor]')) {
```
Comment (willemvb) [inline resources/assets/js/back/app.js]: Not commented out by default anymore? It's easier for us now, but mix fails on first run for newbies

File: resources/assets/js/back/app.js (hunk 6 above)
Comment (willemvb) [reply resources/assets/js/back/app.js]: Could be usefull to quickly 
- check filesize for chunks in cli output
- allocate browser errors to a certain chunk

**Comment by willemvb** [comment]
Nice job sir!

File: package.json (hunk 9 above)
Comment (sebastiandedeyne) [reply package.json]: Good catch, will dump

File: resources/assets/js/back/app.js (hunk 10 above)
Comment (sebastiandedeyne) [reply resources/assets/js/back/app.js]: My bad, needed to give redactor a spin to make sure it works. I'll add it back before merging!

**Pull request merged**